data/modules/CargoRun/CargoRun.lua

```diff
@@ -0,0 +1,1148 @@
+	form:AddOption(l.HOW_SOON_MUST_IT_BE_DELIVERED, 4)
+	form:AddOption(l.WILL_I_BE_IN_ANY_DANGER, 5)
+	form:AddOption(l.COULD_YOU_REPEAT_THE_ORIGINAL_REQUEST, 0)
+	form:AddOption(l.OK_AGREED, 3)
+end
+
+local onDelete = function (ref)
+	ads[ref] = nil
+end
+
+local isEnabled = function (ref)
+	return isQualifiedFor(Character.persistent.player.reputation, ads[ref])
+end
+
+local findNearbyStations = function (station, minDist)
+	local nearbystations = {}
+	for _,s in ipairs(Game.system:GetStationPaths()) do
+		if s ~= station.path then
+			local dist = station:DistanceTo(Space.GetBody(s.bodyIndex))
+			if dist >= minDist then
+				table.insert(nearbystations, { s, dist })
+			end
+		end
+	end
+	return nearbystations
+end
+
+local nearbysystems
+
+local makeAdvert = function (station, freeCargoSpace)
+	local reward, due, location, nearbysystem, dist, nearbystations, cargo, adtext, introtext
+	local risk, wholesaler, pickup
+	local client = Character.New()
+	local urgency = Engine.rand:Number(0, 1)
+	local localdelivery = Engine.rand:Number(0, 1) > 0.5 and true or false
+	local branch = Engine.rand:Integer(1, #custom_cargo)
+	local cargotype = Engine.rand:Integer(1, #custom_cargo[branch])
+
+	if localdelivery then
+		nearbysystem = Game.system
+		nearbystations = findNearbyStations(station, 1000)
+		if #nearbystations == 0 then return nil end
+		if freeCargoSpace ~= nil then
+			cargo = freeCargoSpace == 1 and 1 or Engine.rand:Integer(1, freeCargoSpace)
+		else
+			cargo = Engine.rand:Integer(1, max_cargo)
+		end
+		risk = 0 -- no risk for local delivery
+		wholesaler = false -- no local wholesaler delivery
+		pickup = Engine.rand:Number(0, 1) > 0.75 and true or false
+		location, dist = table.unpack(nearbystations[Engine.rand:Integer(1,#nearbystations)])
+		reward = typical_reward_local + (math.sqrt(dist) / 15000) * (1+urgency) * (1+cargo/max_cargo)
+		due = (4*24*60*60) + (24*60*60 * (dist / (1.49*10^11))) * (1.5 - urgency)
+		if pickup then
+			introtext = Engine.rand:Integer(1, getNumberOf("INTROTEXT_PICKUP_LOCAL"))
+			reward = reward * pickup_factor
+			due = due * pickup_factor + Game.time
+		else
+			introtext = Engine.rand:Integer(1, getNumberOf("INTROTEXT_LOCAL"))
+			due = due + Game.time
+		end
+	else
+		if nearbysystems == nil then
+			nearbysystems = Game.system:GetNearbySystems(max_delivery_dist, function (s) return #s:GetStationPaths() > 0 end)
+		end
+		if #nearbysystems == 0 then return nil end
+		nearbysystem = nearbysystems[Engine.rand:Integer(1,#nearbysystems)]
+		dist = nearbysystem:DistanceTo(Game.system)
+		nearbystations = nearbysystem:GetStationPaths()
+		location = nearbystations[Engine.rand:Integer(1,#nearbystations)]
+		if freeCargoSpace ~= nil then
+			local approxFuelForJump = math.floor(dist / 10)
+			cargo = (freeCargoSpace - approxFuelForJump) <= 1 and 1 or Engine.rand:Integer(1, (freeCargoSpace - approxFuelForJump))
+			introtext = Engine.rand:Integer(1, getNumberOf("INTROTEXT"))
+			wholesaler = false
```

Comment:
Nitpicking: `pickup=false` is set explicitly at several places (which is nice, makes things clear) but not for this condition, which relies on its default behaviour for an unassigned variable.